Sources/Logging/Logging.swift

```diff
@@ -126,6 +111,33 @@ public enum Logging {
     }
 }
 
+extension Logging {
+    public typealias Metadata = [String: MetadataValue]
+
+    public enum MetadataValue {
+        case string(String)
```

Comment:
As we discussed in https://forums.swift.org/t/server-logging-api/18834/58?u=ktoso this would be nice to include the `lazy` value type. 

Either `case lazy(() -> String)` or `case lazy(() -> MetadataValue)` would work.
Perhaps the more general one would serve more use cases than the single one I have in mind... e.g. if constructing the metadata list is expensive etc.

Reply:
I think it's overkill but I'll implement it :)